array.c

```diff
@@ -5067,32 +5067,220 @@ rb_ary_clear(VALUE ary)
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(:X, (-1..1)) # => ["a", "b", "c", "d"]
+ *
+ *  If <tt>range.last</tt> is negative, counts from the end:
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(:X, (0..-2)) # => [:X, :X, :X, "d"]
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(:X, (1..-2)) # => ["a", :X, :X, "d"]
+ *
+ *  If <tt>range.last</tt> and <tt>range.last</tt> are both negative,
+ *  both count from the end of the array:
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(:X, (-1..-1)) # => ["a", "b", "c", :X]
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(:X, (-2..-2)) # => ["a", "b", :X, "d"]
+ *
+ *  ---
+ *
+ *  With no arguments and a block given, calls the block with each index;
+ *  replaces the corresponding element with the block's return value:
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill { |index| "new_#{index}" } # => ["new_0", "new_1", "new_2", "new_3"]
+ *
+ *  ---
  *
- *  The first three forms set the selected elements of +self+ (which
- *  may be the entire array) to +obj+.
+ *  With argument +start+ and a block given, calls the block with each index
+ *  from offset +start+ to the end; replaces the corresponding element
+ *  with the block's return value:
  *
- *  A +start+ of +nil+ is equivalent to zero.
+ *  If start is in range (<tt>0 <= start < ary.size</tt>),
+ *  replaces from offset +start+ to the end:
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(1) { |index| "new_#{index}" } # => ["a", "new_1", "new_2", "new_3"]
  *
- *  A +length+ of +nil+ is equivalent to the length of the array.
+ *  If +start+ is too large(<tt>start >= ary.size</tt>), does nothing:
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(4) { |index| fail 'Cannot happen' } # => ["a", "b", "c", "d"]
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(4) { |index| fail 'Cannot happen' } # => ["a", "b", "c", "d"]
+ *
+ *  If +start+ is negative, counts from the end:
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(-2) { |index| "new_#{index}" } # => ["a", "b", "new_2", "new_3"]
+ *
+ *  If start is too small (<tt>start <= -ary.size</tt>, replaces all elements:
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(-6) { |index| "new_#{index}" } # => ["new_0", "new_1", "new_2", "new_3"]
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(-50) { |index| "new_#{index}" } # => ["new_0", "new_1", "new_2", "new_3"]
+ *
+ *  ---
+ *
+ *  With arguments +start+ and +length+, and a block given,
+ *  calls the block for each index specified by start length;
+ *  replaces the corresponding element with the block's return value.
+ *
+ *  If +start+ is in range, replaces +length+ elements beginning at offset +start+:
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(1, 1) { |index| "new_#{index}" } # => ["a", "new_1", "c", "d"]
+ *
+ *  If start is negative, counts from the end:
+ *    a = ['a', 'b', 'c', 'd']
+ *    a.fill(-2, 1) { |index| "new_#{index}" } # => ["a", "b", "new_2", "d"]
+ *
+ *  If +start+ is large(<tt>start >= ary.size</tt>), extends +self+ with +nil+:
```

Comment:
How about one extra space before `(`:
```suggestion
 *  If +start+ is large (<tt>start >= ary.size</tt>), extends +self+ with +nil+:
```